### DISCUSSION TITLE
Native Network Encryption(NNE) Error: NJS-500: Connection to the Oracle Database was broken NJS-532: connection to host...received end-of-file

### DISCUSSION THREAD

**Discussion by rhettjay** [oracle/node-oracledb#1553] [Q&A] [answered]
I've tried utilizing the easy connect method to connect through a CMAN gateway, but I get the following error
`Error: NSJ-500: connection to the Oracle Database was broken 
NJS-532: connection to host {hostname-redacted} port {port-redacted} received end-of-file on communication channel (CONNECTION_ID={id-redacted})
    at NTTCP.checkErr (/example/node_modules/oracledb/lib/thin/sqlnet/ntTcp.js:305:23)
    at NTTCP.receive (/example/node_modules/oracledb/lib/thin/sqlnet/ntTcp.js:446:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async NetworkSession._recvPacket (/example/node_modules/oracledb/lib/thin/sqlnet/networkSession.js:376:22)
    at async NetworkSession.recvPacket (/example/node_modules/oracledb/lib/thin/sqlnet/networkSession.js:441:12)
    at async ReadPacket.waitForPackets (/example/node_modules/oracledb/lib/thin/protocol/packet.js:288:20)
    at async Protocol._decodeMessage (/example/node_modules/oracledb/lib/thin/protocol/protocol.js:81:11)
    at async Protocol._processMessage (/example/node_modules/oracledb/lib/thin/protocol/protocol.js:148:9)
    at async ThinConnectionImpl.connect (/example/node_modules/oracledb/lib/thin/connection.js:533:7)
    at async Object.getConnection (/example/node_modules/oracledb/lib/oracledb.js:628:3) {
  code: 'NJS-500'
}`

I'm running pretty much exactly what is in examples/connect.js with a dbconfig.js. The only adjustments being my test username and password in addition to a connection string that routes through a cman gateway. Should this already be possible or are there further adjustments I need to make in order to connect to a database through a cman?

**Comment by sharadraju** [comment]
Hi @jay-bulk, Thank you for trying our new Thin mode. 
Were you able to connect to CMAN using the same connect string with the previous release of node-oracledb?

**Reply by rhettjay** [reply]
Hi @sharadraju, yes. I don't know that it's a cman issue though, a couple of my colleagues have been testing it as well and they are running into it as well outside the cman. As of right now we're undetermined what the discrepancy is. We're all testing with the same script in examples but against our stage servers. Is there a debug setting I'm unaware of that might offer up an error closer to the database server? We're breaking before authentication occurs so it appears to be specific to the connection string or its inherited settings.

**Reply by cjbj** [reply]
@jay-bulk 
- CMAN usage should work.
- can you confirm you are using Thin mode (i.e. no `initOracleClient()` calls in your code), and if so, can you check that [Thick mode](https://node-oracledb.readthedocs.io/en/latest/user_guide/initialization.html#enabling-node-oracledb-thick-mode) continues to work like it did with 5.5?
- If you are using wallets, note the connection syntax differs between Thin & Thick modes, and that Thin mode doesn't load sqlnet.ora files. Instead properties can be set in the getConnection/createPool calls.
- There is a low-level trace dump in Thin mode that can be enabled, see [here](https://node-oracledb.readthedocs.io/en/latest/user_guide/tracing.html#low-level-node-oracledb-driver-tracing).  Sharing actual connection strings and logs with us would help. Email to @sharadraju or me is fine, see our profiles.

**Reply by cjbj** [reply, marked as answer]
One more question from the developers: do you have [Native Network Encryption](https://node-oracledb.readthedocs.io/en/latest/user_guide/connection_handling.html#native-network-encryption) enabled?  Thin mode doesn't/won't support this.

**Reply by krismohan** [reply]
What is the database version?

**Reply by rhettjay** [reply]
@krismohan Oracle Enterprise 19.0.
@cjbj checking on the native network encryption now.

**Reply by rhettjay** [reply]
@cjbj we do have native network encryption enabled. I've tested with it Native Network Encryption set to requested and the connection worked. Then once more with it re-enabled or "required" and it broke again. Looks like we found my issue. Thanks all!

**Reply by rhettjay** [reply]
I've renamed this thread to improve clarity for those who encounter the error down the road.

**Reply by PauladsBastos** [reply]
Hi @rhettjay How did you resolved this issue?

**Reply by rhettjay** [reply]
Used the [thick client](https://node-oracledb.readthedocs.io/en/latest/user_guide/appendix_a.html#appendix-a-the-node-oracledb-thin-and-thick-modes) when included in a docker container there are a few extra steps.
See [thick mode initialization](https://node-oracledb.readthedocs.io/en/latest/user_guide/initialization.html#enabling-node-oracledb-thick-mode) and [docker example](https://node-oracledb.readthedocs.io/en/latest/user_guide/installation.html#example-application-in-docker) for applicable documentation.

**Reply by cjbj** [reply]
Also see https://blogs.oracle.com/opal/post/part-1-docker-for-oracle-database-applications-in-nodejs-and-python

**Reply by PauladsBastos** [reply]
Thank You for Your Help! 🙌
